base/uuid.jl

```diff
@@ -31,6 +31,11 @@ end
 
 UInt128(u::UUID) = u.value
 
+let
+    uuid_hash_seed = UInt === UInt64 ? 0xd06fa04f86f11b53 : 0x96a1f36d
+    hash(uuid::UUID, h::UInt) = hash(uuid_hash_seed, hash(convert(NTuple{2, UInt64}, uuid), h))
```

Comment:
Doesn't this just create a new local function? :)

Reply:
Oh gosh, good catch 😄.

Reply:
I.e., it looks like this change follows the pattern of the following `let` blocks, in which case `hash` is missing a qualification that would lift this `hash` method from the `let`-block's scope? :)

Reply:
I am not sure you can overload a function from inside a `let` block. The easiest solution is probably just to make `uuid_hash_seed` a global constant.

Reply:
The code just a bit below uses that pattern. See also e.g.
```
julia> struct FluffType end

julia> let
       Base.hash(::FluffType) = "fluffy"
       end

julia> hash(FluffType())
"fluffy"
```

Reply:
Oops, thanks.